File: python/stack/10828.py
from collections import deque

n = int(input())

stack = deque()

while n > 0:
    string = input()
    cmd = string.split()
    if cmd[0] == 'push':
        stack.append(cmd[1])
    elif cmd[0] == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)
    elif cmd[0] == "empty":
        if stack:
            print(0)
        else:
            print(1)
    elif cmd[0] == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif cmd[0] == "size":
        print(len(stack))

    n -= 1

File: algorithm_go/src/stack/10828/10828.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"stack"
	"strconv"
	"strings"
)

func main(){
	r := bufio.NewReader(os.Stdin)
	w := bufio.NewWriter(os.Stdout)

	var n int
	var cmd string

	s := stack.Stack(100)

	fmt.Fscanln(r, &n)

	for i := 0; i< n; i++ {

		line, _, _ := r.ReadLine()

		parts := strings.Split(string(line), " ")
		cmd = parts[0]

		if cmd == "push"{
			number, _ := strconv.Atoi(parts[1])
			s.Push(number)
		} else if cmd == "pop" {
			fmt.Fprintln(w, s.Pop())
		} else if cmd == "size" {
			fmt.Fprintln(w, s.Size())
		} else if cmd == "top" {
			fmt.Fprintln(w, s.Top())
		} else if cmd == "empty" {
			fmt.Fprintln(w, s.Empty());
		}
	}
	w.Flush()


}
